nsxt/resource_nsxt_edge_transport_node.go

```diff
@@ -1474,7 +1699,12 @@ func setHostSwitchSpecInSchema(d *schema.ResourceData, spec *data.StructValue, n
 				elem["host_switch_profile"] = profiles
 			}
 			var err error
-			elem["ip_assignment"], err = setIPAssignmentInSchema(sw.IpAssignmentSpec)
+			if sw.IpAssignmentSpec != nil {
+				elem["ip_assignment"], err = setIPAssignmentInSchema(sw.IpAssignmentSpec)
+			}
+			if sw.Ipv6AssignmentSpec != nil {
+				elem["ipv6_assignment"], err = setIPv6AssignmentInSchema(sw.Ipv6AssignmentSpec)
+			}
 			if err != nil {
```

Comment:
I'd rather check err after 1073, 1076 lines - as err from `setIPAssignmentInSchema` could be overwritten by `setIPv6AssignmentInSchema`

Reply:
Thanks!